src/CommunicationLayer/JsonReceiver/JsonReceiver.cpp

```diff
@@ -38,7 +38,8 @@ JsonReceiver::JsonReceiver(I_CommDevice& inputDevice,
                            MotorDetailsPopulator& motorDetailsPopulator,
                            MotorFaultsPopulator& motorFaultsPopulator,
                            I_CommunicationsMonitoringService& communicationsMonitoringService)
-    : batteryPopulator_(batteryPopulator)
+    : inputDevice_(inputDevice)
```

Comment:
the JsonReceiver doesn't need to keep track of the input device, it just hooks up the signals and that's it.

Reply:
Should the container hook up the input device and JSONReceiver then?